### PR TITLE
fix: add Optimizely builder option for client-engine info

### PR DESCRIPTION
**Pull request by jaeopt**
## Summary
- Add an option to specify client-engine version (android-sdk, etc) in the Optimizely builder.

## Test plan
- Add a unit test for builder with client-engine version.

## Issues
- OASIS-8188

### CODE REVIEW

File: core-api/src/main/java/com/optimizely/ab/Optimizely.java [hunk 1]
```diff
@@ -1519,6 +1516,12 @@ public Builder withUserProfileService(UserProfileService userProfileService) {
             return this;
         }
 
+        public Builder withClientInfo(EventBatch.ClientEngine clientEngine, String clientVersion) {
```
Comment (msohailhussain) [inline core-api/src/main/java/com/optimizely/ab/Optimizely.java:1519]: please add documentation.

File: core-api/src/main/java/com/optimizely/ab/event/internal/BuildVersionInfo.java [hunk 2]
```diff
@@ -30,13 +30,25 @@
...
 public final class BuildVersionInfo {
 
     private static final Logger logger = LoggerFactory.getLogger(BuildVersionInfo.class);
 
+    @Deprecated
     public final static String VERSION = readVersionNumber();
 
+    // can be overridden by other wrapper client (android-sdk, etc)
+
+    private static String clientVersion = readVersionNumber();
```
Comment (msohailhussain) [inline core-api/src/main/java/com/optimizely/ab/event/internal/BuildVersionInfo.java:42]: final?

File: core-api/src/test/java/com/optimizely/ab/OptimizelyBuilderTest.java [hunk 3]
```diff
@@ -195,4 +203,33 @@ public void withDefaultDecideOptions() throws Exception {
...
+
+        verify(eventHandler, timeout(5000)).dispatchEvent(argument.capture());
+        assertEquals(argument.getValue().getEventBatch().getClientName(), "java-sdk");
+        assertEquals(argument.getValue().getEventBatch().getClientVersion(), BuildVersionInfo.getClientVersion());
+
+        // override client-engine info
+
+        eventHandler = mock(EventHandler.class);
+        optimizely = Optimizely.builder(validConfigJsonV4(), eventHandler)
+            .withClientInfo(EventBatch.ClientEngine.ANDROID_SDK, "1.2.3")
```
Comment (msohailhussain) [inline core-api/src/test/java/com/optimizely/ab/OptimizelyBuilderTest.java:226]: isn't it confusing to check "android-sdk" client engine.

**Review by msohailhussain — APPROVED** [review]
lgtm

File: core-api/src/main/java/com/optimizely/ab/event/internal/BuildVersionInfo.java (hunk 2 above)
Comment (jaeopt) [reply core-api/src/main/java/com/optimizely/ab/event/internal/BuildVersionInfo.java:42]: Unlike the old VERSION, this private variable can be changed with Optimizely.Builder.

File: core-api/src/test/java/com/optimizely/ab/OptimizelyBuilderTest.java (hunk 3 above)
Comment (jaeopt) [reply core-api/src/test/java/com/optimizely/ab/OptimizelyBuilderTest.java:226]: Can you clarify this comment? Not clear to me.

**Pull request merged**